layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -184,6 +184,58 @@ Dedicated (locked) windows are left untouched."
   (interactive "p")
   (spacemacs/rotate-windows (* -1 count)))
 
+;; Note that this duplicates code from select-window-by-number, ideally should
+;; upstream this function into windows.el
+(defun spacemacs/get-window-by-number (i)
+  (let ((windows (car (gethash (selected-frame) window-numbering-table)))
+        window)
+    (if (and (>= i 0) (< i 10)
+             (setq window (aref windows i)))
+        window
+      (error "No window numbered %s" i)))
+  )
```

Comment:
Didn't notice in previous version, but elisp convention is that closing parentheses don't deserve their own line. From the [elisp manual](https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html):
> Don’t make a habit of putting close-parentheses on lines by themselves; Lisp programmers find this disconcerting.

It's not crucial, though.